Tests/SwiftSyntaxBuilderTest/ProtocolDeclTests.swift

```diff
@@ -0,0 +1,29 @@
+import XCTest
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+final class ProtocolDeclTests: XCTestCase {
+  func testProtocolDecl() {
+    let returnClause = ArrayType(elementType: "DeclSyntax")
+    let input = ParameterClause(parameterListBuilder: {
+      FunctionParameter(firstName: .identifier("format"), colon: .colon, type: "Format", trailingComma: .comma, attributesBuilder: {})
+      FunctionParameter(firstName: .identifier("leadingTrivia"), colon: .colon, type: OptionalType(wrappedType: "Trivia"), attributesBuilder: {})
```

Comment:
Is the `attributesBuilder` argument necessary? Shouldn’t it be defaulted?

Reply:
It should yes. 
But the problem is that we then get `Ambiguous use of 'init'`.

I think we should, somehow, solve this as there is many `init` methods that look like each other so we need to do things like this to avoid the error. Now it works now, but I think it makes some noice when using it.

This is a long shot, and I'm not sure if it is technically possible, but I'll try to explain. 

It would be nice if we could remove this type of `init` methods, where the `@resultBuilder` is the only difference.
This will mean, in this case, that our `AttributeListBuilder` need to conform to `ExpressibleAsAttributeList`.
And then we could things like `FunctionParameter(modifiers: { TokenSyntax.public }, firstName: .identifier("format"), colon: .colon, type: "Format", trailingComma: .comma)` or 
`FunctionParameter(modifiers: TokenSyntax.public, firstName: .identifier("format"), colon: .colon, type: "Format", trailingComma: .comma)`. 

I think we before have talked about that the `@resultBuilders` should conform to some of the protocols.

Reply:
I see, makes sense 👍 I’m good with merging this as-is.

As to your other question: 
If I understand you correctly, you are saying that you’d like to remove the second initializer listed in the code snippet and make `AttributeListBuilder` conform to `ExpressibleAsAttributeList`. In that case you expect to be able to use the `AttributeListBuilder` result builder wherever an `ExpressibleAsAttributeList` is expected, right? 
If this is what you’re suggesting, I’m pretty sure it won’t work. I think the result builder needs to be declared using `@AttributeListBuilder` on the parameter or otherwise the type checker won’t try to apply it.